spec/support/similarity.rb

```diff
@@ -0,0 +1,31 @@
+module Similarity
+  def self.test(lexer_class)
+    # state_defintions is an InheritableHash, so we use `own_keys` to
+    # exclude states inherited from superclasses
+    state_names = Set.new(lexer_class.state_definitions.own_keys)
+
+    candidates = Rouge::Lexer.all.select do |x|
+      # we can only compare to RegexLexers which have state_definitions
+      next false unless x < Rouge::RegexLexer
+
+      # don't compare a lexer to itself or any subclasses
+      next false if x <= lexer_class
+
+      true
+    end
+
+    max_score = 1
```

Comment:
It's possibly overengineering this but if we create a constant, `Similarity::MAX` we can then set `max_score` to this value and use it in the Rake task. That'll make it trivial to update if we decide we want a higher maximum score (eg. because lots of lexers involve a `:root` state, a `:whitespace` state, etc).

Reply:
yeah! depends on the use case - this is only informative for now.

Reply:
I was imagining that we'd have some kind of bot that comments on new lexer PRs, and we can simply choose to ignore it.

Reply:
(basically it will still be up to the maintainer to decide whether it really is a copy-paste job or not)

Reply:
Yeah, I'm not sure about the mechanics of automating this; I still need to look into that.